WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsCollectionViewFlowLayout.swift

```diff
@@ -0,0 +1,204 @@
+
+    override open var collectionViewContentSize: CGSize {
+        return contentSize
+    }
+
+    override func prepare() {
+        guard let collectionView = collectionView else {
+            return
+        }
+
+        let contentInsets = collectionView.contentInset
+        let isRTL = layoutDirection == .rightToLeft
+
+        // The current row used to calculate the y position and the total content height
+        var currentRow: CGFloat = 0
+
+        // Keeps track of the previous items frame so we can properly calculate the current item's x position
+        var previousFrame: CGRect = .zero
+
+        let numberOfItems: Int = collectionView.numberOfItems(inSection: 0)
+
+        for item in 0 ..< numberOfItems {
+            let indexPath: IndexPath = IndexPath(row: item, section: 0)
+
+            let itemSize = sizeForItem(at: indexPath)
+            var frame: CGRect = CGRect(origin: .zero, size: itemSize)
+
+            if item == 0 {
+                let minX: CGFloat = isRTL ? maxContentWidth - frame.width : 0
+                frame.origin = CGPoint(x: minX, y: contentInsets.top)
+            } else {
+                if isRTL {
+                    frame.origin.x = previousFrame.minX - itemSpacing - frame.width
+                } else {
+                    frame.origin.x = previousFrame.maxX + itemSpacing
+                }
+
+                // If the new X position will go off screen move it to the next row
+                let needsNewRow = isRTL ? frame.origin.x < 0 : frame.maxX > maxContentWidth
+                if needsNewRow {
+                    frame.origin.x = isRTL ? (maxContentWidth - frame.width) : 0
+                    currentRow += 1
+                }
+                frame.origin.y = currentRow * (cellHeight + itemSpacing) + contentInsets.top
+            }
+
+            let attribute = UICollectionViewLayoutAttributes(forCellWith: indexPath)
+            attribute.frame = frame
+
+            layoutAttributes.append(attribute)
+
+            previousFrame = frame
+        }
+
+        // Update content size
+        contentSize.width = maxContentWidth
+        contentSize.height = currentRow * (cellHeight + itemSpacing) + contentInsets.top + contentInsets.bottom
+    }
+
+    override func invalidateLayout() {
+        contentSize = .zero
+        layoutAttributes = []
+
+        super.invalidateLayout()
+    }
+
+    /// Get the size for the given index path from the delegate, or the default item size
+    /// - Parameter indexPath: index path for the item
+    /// - Returns: The width for the cell either from the delegate or the itemSize property
+    private func sizeForItem(at indexPath: IndexPath) -> CGSize {
+        guard
+            let collectionView = collectionView,
+            let delegate = collectionView.delegate as? UICollectionViewDelegateFlowLayout,
+            delegate.responds(to: #selector(delegate.collectionView(_:layout:sizeForItemAt:)))
```

Comment:
I don't think there's a need to check for respond(to: #selector) here 
Also please consider avoiding the forced unwrap in line 103 and add the size to the guard instead. 

So this method could be:
``` 
private func sizeForItem(at indexPath: IndexPath) -> CGSize {
        guard
            let collectionView = collectionView,
            let delegate = collectionView.delegate as? UICollectionViewDelegateFlowLayout,
            let size = delegate.collectionView?(collectionView, layout: self, sizeForItemAt: indexPath)
        else {
            return CGSize(width: itemSize.width, height: cellHeight)
        }
        return CGSize(width: size.width, height: cellHeight)
    }
```

Reply:
D'oh. There's that Objective-C thinking creeping in again. Done: https://github.com/wordpress-mobile/WordPress-iOS/pull/14389/commits/316ae18bc4ad7ffe05e4314abc870fbd82e2df95